src/qibolab/instruments/qm.py

```diff
@@ -32,14 +32,16 @@
 
 # TODO: Split the contents of this module to multiple files
 
-PortType = Tuple[str, int]
+PortId = Tuple[str, int]
 """Type for port definition, for example: ("con1", 2)."""
+IQPortId = Union[Tuple[PortId], Tuple[PortId, PortId]]
```

Comment:
Just a last tiny remark on this: in the case of one port, do you really want the singleton `Tuple[PortId]` (e.g. `(1,)`), or would you rather get directly a `PortId`?

I have no opinion about this, just to make sure the type is the correct one.

Reply:
I think I am using the singleton because later in https://github.com/qiboteam/qibolab/blob/a2af4f886f7b919549294ea884e327a3701e0e8a/src/qibolab/instruments/qm.py#L748
it is more convenient as `*port_name` handles the creation of ports for both cases (singleton and pair). If I change to just `PortId` I think I would have to seperate the two cases, either with try-except or some if.

That is not a very strong argument though, and I have not strong opinion either.

On that note, while implementing the `Port`s for the other instruments (see new PRs), I was thinking if we should be more strict regarding the port naming/creation and force a uniform interface. For example have a `name` attribute for all `Port`s. Now I leave it up to each driver to decide, mostly for convenience.

Reply:
> That is not a very strong argument though, and I have not strong opinion either.

For me, it is quite a strong motivation: fewer special cases, always good :)

> On that note, while implementing the `Port`s for the other instruments (see new PRs), I was thinking if we should be more strict regarding the port naming/creation and force a uniform interface. For example have a `name` attribute for all `Port`s. Now I leave it up to each driver to decide, mostly for convenience.

If it's something we could make use in some common code, for sure it's a great idea (at least we minimize the duplication). Otherwise, better to keep the interface minimal, to reduce the burden on the implementor.

There is always a trade-off: a very detailed interface is more effective, but more of a pain to implement and less general, while a loose one is easier to implement but less useful - we should try to enforce all distinctive features, but no more.
If you think about Python protocols, like `Iterable`/`Iterator` or `Sequence` and similar, they are extremely poor, but extremely general, and thus useful. But we're not writing something so general, so for sure this interface will have more features (how many more... again: how many are really common).